### DISCUSSION TITLE
What is event activity types marked this pull request as draft in Github action?

### DISCUSSION THREAD

**Discussion by anteatera** [community/community#27022] [Actions]
<p><img src="https://user-images.githubusercontent.com/64678460/181101189-502148d7-b8f3-49ca-b696-7cd0c104c9f7.png" alt="Screen Shot 2020-05-02 at 15.54.13.png" data-base62-sha1="oTdFz9yYiHmQeWCHqhAJUSqHeeL" width="478" height="214"></p>
<p>Hello! I have one question about github action events that trigger.</p>
<p>what is event activity types to <code>marked this pull request as draft</code> in Github action?</p>

**Comment by anteatera** [comment]
<p>I found this reply.</p>
<blockquote>
<p>According to <a href="https://developer.github.com/changes/2019-02-14-draft-pull-requests/" rel="nofollow noopener">the blog post introducing this support</a>, it appears that creating draft pull requests is limited to the v4 GraphQL API and the v3 REST API only supports draft indicators in API responses</p>
</blockquote>
<p><a href="https://github.community/t/create-draft-pull-request-from-the-api/13788">https://github.community/t5/GitHub-API-Development-and/Create-Draft-Pull-Request-from-the-API/td-p/19288</a></p>
<p>any suggestions in GraphQL API v4 or REST API v3?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/anteatera">@anteatera</a> ,</p>
<p>The draft pull request could be create by:</p>
<ol>
<li>In pull request UI:</li>
</ol>
<p><img src="https://user-images.githubusercontent.com/52772761/181101191-0bb8ec81-7fd2-4d7f-a9b9-f836f64fda52.png" alt="draft_pull_request1.png" data-base62-sha1="o5opqEpPtqRcEzhyz2EuSCGyQ4D" width="436" height="214"></p>
<ol start="2">
<li>Convert the existing pull_request to draft.</li>
</ol>
<p><img src="https://user-images.githubusercontent.com/52772761/181101192-23faefe5-9574-4b1e-a2cf-6fb52918e65a.png" alt="draft_pull_request2.png" data-base62-sha1="rVYPbDpx3FdWRiH4IfKrDid2BKl" width="324" height="120"></p>
<ol start="3">
<li>From <a href="https://developer.github.com/v3/pulls/#create-a-pull-request" rel="nofollow noopener">API</a> command: add “draft”:  “true” in the request body.</li>
</ol>
<p><img src="https://user-images.githubusercontent.com/52772761/181101193-8fb2e29f-a643-46c0-90cd-063becb0ec16.png" alt="draft_pull_request3.png" data-base62-sha1="l4WlFsn2sZeCbrSAgCEnzV9yd0d" width="689" height="346"></p>
<p><strong>There is no specific event activity types for draft pull request.</strong></p>
<p>When you create new draft pull request(method 1&amp;3), workflow can be triggered with event ‘on: pull_request’.</p>
<p>When you change existing pull request to ‘draft’(method 2), there is no workflow triggered.</p>
<p>You can get the draft info from github context {{ github.event.pull_request.draft }}, as an alternative, you can use if expression to check whether it’s a draft pull_request and determine the job execution accordingly. Code sample as below:</p>
<pre><code>on: [pull_request]

jobs:
  job1:
    if: github.event.pull_request.draft == true
    runs-on: [ubuntu-latest]
    steps:
...
</code></pre>
<p>Hope it’s clear and helpful.</p>

**Comment by anteatera** [comment]
<p>You’ve been very helpful. Thank you! <a class="mention" href="/weide-zhou">@weide-zhou</a></p>
